infrastructure/templates/public-api/components/functionApp.bicep

```diff
@@ -195,9 +207,20 @@ var commonSiteProperties = {
     netFrameworkVersion: '8.0'
     linuxFxVersion: appServicePlanOS == 'Linux' ? 'DOTNET-ISOLATED|8.0' : null
     keyVaultReferenceIdentity: keyVaultReferenceIdentity
+    publicNetworkAccess: publicNetworkAccessEnabled ? 'Enabled' : 'Disabled'
+    ipSecurityRestrictions: publicNetworkAccessEnabled && length(firewallRules) > 0 ? firewallRules : null
+    ipSecurityRestrictionsDefaultAction: publicNetworkAccessEnabled && length(firewallRules) > 0 ? 'Deny' : 'Allow'
+    scmIpSecurityRestrictions: [
+      {
+        ipAddress: 'Any'
+        action: 'Allow'
+        priority: 2147483647
+        name: 'Allow all'
+        description: 'Allow all access'
+      }
+    ]
```

Comment:
I believe it's IP restrictions to determine who's able to see the part of the stack that's responsible for performing deploys to the Function App / App Service.  So ideally when we have static IPs for our runners, we'd be able to whitelist them so that only the pipeline would be able to deploy code to these resources.  Might also control who's able to access the Kudu site that accompanies each Function App / App Service host (the one you can access through `Advanced tools` in Azure Portal.

I've stuck a comment on:

```
    // TODO EES-5446 - this setting controls access to the deploy site for the Function App.
    // This is currently the default value, but ideally we would lock this down to only be accessible
    // by our runners and certain other whitelisted IP address ranges (e.g. trusted VPNs).
```